strapi/strapi
# Fix documentation plugin + dynamic security middleware plugin endpoints

### What does it do?

1. Modifies the `security` middleware so that plugin-related CSP directives are enforced on the GraphQL and Documentation plugin endpoints even when they are customised.
2. Updates all routes related to the Documentation plugin on register to properly set the correct endpoint, so as to solve #14020.
3. Remove the `README.md` file from the Documentation plugin's package.

### Why is it needed?

1. The `security` middleware was only enforcing the CSP directives on endpoints starting with `/graphql` and `/documentation`.
2. Solves #14020.
3. Configuring the documentation plugin through the `settings.json` file in the `extensions` folder does not work as expected, whereas configuring the `documentation.config` object in the `config/plugins.js` file does. Since most of the information in the README still refers to the first method of handling customisations, I propose we nuke it similarly to the GraphQL plugin's README and forward users to the Strapi documentation.

### How to test it?

Try changing the endpoints for GraphQL and Documentation and notice how they work without issues.

NOTE: The Strapi documentation should also be updated if this PR is accepted.
